.github/workflows/auto-update-gemfiles.yml

```diff
@@ -0,0 +1,169 @@
+name: Auto-update Gemfiles
+
+# This action cannot be skipped altogether because it is a mandatory status check.
+# Instead we conditionally skip it at job level, instead of workflow level.
+on:
+  schedule:
+    - cron: '0 0 * * 0' # Every Sunday at midnight
+  workflow_dispatch:
+
+# Ensure obsolete job is cancelled if another commit is pushed to the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    permissions:
```

Comment:
Do we need those permissions?

Reply:
Thanks Tony, I _think_ it needs `contents: write` for artifact uploading, but I may be wrong. @quinna-h could we try removing both `actions: read` and `contents: write` and seeing if it still works?

Reply:
I was able to upload artifacts without them: https://github.com/DataDog/dd-trace-rb/pull/3926/files#diff-50517d6517bd2924edda04f5d4e68e631ddc36d92cff5720e54fbde4f2a46c50R59

Reply:
It seems to work without `actions: read` and `contents: write` , but I think we still need `contents: read`